### ISSUE TITLE
Read/write roundtrip produces inconsistent spectral axis after changing reference frame

### ISSUE THREAD

**Issue by astrofle** [GreenBankObservatory/dysh#434]
**Describe the bug**
After reading a `Spectrum` from a fits file written by dysh, I'd expect the read-in `Spectrum` to behave the same as the original `Spectrum`, but it does not.

**How to Reproduce**
```Python
from dysh.spectra import Spectrum

s = Spectrum.fake_spectrum()
s.write("test_frame.fits", format="fits", overwrite=True)
s1 = Spectrum.read("test_frame.fits", format="fits")
print(s1.spectral_axis - s.spectral_axis) # So far so good.
# This breaks:
s.set_frame("lsrk")
s1.set_frame("lsrk")
print(s1.spectral_axis - s.spectral_axis) # Difference is not zero!
```

**Environment**
- Dysh version 0.4.0
- Python version 3.10
- OS

**Comment by mpound** [comment]
The observer attributes on the spectral axes are slightly different.  I wonder if how this affects `set_frame`.

```from dysh.spectra import Spectrum

s = Spectrum.fake_spectrum()
s.write("test_frame.fits", format="fits", overwrite=True)
s1 = Spectrum.read("test_frame.fits", format="fits")
print(s.spectral_axis.observer)
print(s1.spectral_axis.observer)
```
<ITRS Coordinate (obstime=2021-02-10T07:38:37.500, location=(0., 0., 0.) km): (x, y, z) in m
    (882593.9465029, -4924896.36541728, 3943748.74743984)
 (v_x, v_y, v_z) in km / s
    (0., 0., 0.)>

<ITRS Coordinate (obstime=2021-02-10T07:38:37.500, location=(0., 0., 0.) km): (x, y, z) in m
    (882590.62036715, -4924873.54273607, 3943729.15615663)
 (v_x, v_y, v_z) in km / s
    (0., 0., 0.)>

**Comment by astrofle** [comment]
Good catch. That explains the difference. When reading a FITS file, the observer location is set from the meta attributes (meta["SITELONG"], meta["SITELAT"], meta["SITEELEV"]), while `fake_spectrum` uses the GBT coordinates defined in `dysh.coordinates`.

Now the question is what to do. The difference seems small enough that maybe it makes sense to just use the GBT coordinates as defined in the SDFITS files.
